### ISSUE TITLE
[settings] in single folder setting, both workspaceValue and workspaceFolderValue are set 

### ISSUE THREAD

**Issue by aeschli** [Microsoft/vscode#34386]
- open VSCode on a folder
- have a workspace setting, e.g.
```
{
        "json.schemas": [{
            "fileMatch": ["view.json"],
            "url": "http://json.schemastore.org/bower"
        }]
}
```
- in the extension host access the settings:
```ts
	let folders = workspace.workspaceFolders;
	if (folders) {
		folders.forEach(folder => {
			let jsonConfig = workspace.getConfiguration('json', folder.uri);
			let schemaConfigInfo = jsonConfig.inspect<JSONSchemaSettings[]>('schemas');
```
Observe that in `schemaConfigInfo` both `workspaceValue` and `workspaceFolderValue` are set.
Only one of them should be set, likely `workspaceFolderValue`. `workspaceValue` should be undefined.

I need this to find out if there are actually folder specific settings. The folder == 1 case is of course a bit special.

![image](https://user-images.githubusercontent.com/6461412/30434291-2eb32cf8-9967-11e7-91d8-692ba7e778bb.png)

**Comment by sandy081** [comment]
@aeschli I see your concern. Ideally in this case, `workspaceValue` should be `undefined` but this will break the API. Marking `workspaceFolderValue` as undefined looks wrong to me. 

Happy to have suggestions.

**Comment by aeschli** [comment]
Yes, I guess If you want to be backward compatible, then having `workspaceFolderValue` undefined is the right thing to do for the case that VSCode was opened on a folder. If VSCode was opened on a workspace with one folder, then 'workspaceFolderValue' should be set and 'workspaceValue' undefined.

**Comment by aeschli** [comment]
The issue is still there but I've worked around it.
